src/Symfony/Component/Security/Core/Exception/AccessDeniedException.php

```diff
@@ -57,4 +63,19 @@ public function setSubject($subject)
     {
         $this->subject = $subject;
     }
+
+    /**
+     * Sets an access decision and appends the denied reasons to the exception message.
+     */
+    public function setAccessDecision(AccessDecision $accessDecision)
```

Comment:
this cannot be stateful IMHO

Reply:
What do you suggest ?

Reply:
We could make it stateless and have a specific exception render for AccessDeniedException. WDYT ?